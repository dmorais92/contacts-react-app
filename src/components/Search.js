import React, { PureComponent } from "react";
import PropTypes from "prop-types";
import cx from "../utils/classnames";
import Icon from "./common/Icon";

import "./Search.scss";

class Search extends PureComponent {
  static propTypes = {
    classes: PropTypes.array,
    onChange: PropTypes.func
  };

  static defaultProps = {
    classes: []
  };

  render() {
    const { classes, onChange, ...otherProps } = this.props;
    const classNames = [...classes];

    return (
      <div className="SearchContainer">
        <div className="search-label">
          <Icon icon="search" size="xs" />
          <h4>Search</h4>
        </div>
        <input
          className={`Search${cx(classNames)}`}
          onChange={e => onChange(e)}
          {...otherProps}
        />
      </div>
    );
  }
}

export default Search;
