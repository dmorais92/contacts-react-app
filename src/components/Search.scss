@import "../variables.scss";
.SearchContainer {
  display: flex;
  flex-flow: column nowrap;
  align-items: center;
  margin-bottom: 1vh;
  font-family: 'Roboto';
  font-weight: bold;

  .Search {
    border: 1px solid $default-font-color;
    border-radius: 4px;
    padding: 1vh 1vw;
    width: 30vw;
    box-shadow: $button-box-shadow;
    background: #fff;
    &:hover {
      cursor: pointer;
    }
    &:focus {
      border: 2px solid $primary-border-color;
      outline: none;
    }
  }

  .search-label{
    display: flex;
    flex-flow: row wrap;
    margin: 0.5vh;
    h4 {
      margin: 0;
      width: fit-content;
    }
    .Icon {
      color: $default-font-color;
      margin: 0 10px;
    }
  }
}
