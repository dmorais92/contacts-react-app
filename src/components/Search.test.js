import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Search from './Search';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Search />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders multiple children', () => {
  const component = renderer.create(
    <Search>
      <h1>Testing with multiple children</h1>
      A short paragraph
    </Search>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders with extra classes', () => {
  const component = renderer.create(
    <Search classes={["an", "array", "of classes"]}/>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});