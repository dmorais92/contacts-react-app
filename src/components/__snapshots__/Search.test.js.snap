// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders multiple children 1`] = `
<div
  className="SearchContainer"
>
  <div
    className="search-label"
  >
    <i
      className="Icon fas fa-search xs primary"
    />
    <h4>
      Search
    </h4>
  </div>
  <input
    className="Search"
    onChange={[Function]}
  >
    <h1>
      Testing with multiple children
    </h1>
    A short paragraph
  </input>
</div>
`;

exports[`renders with extra classes 1`] = `
<div
  className="SearchContainer"
>
  <div
    className="search-label"
  >
    <i
      className="Icon fas fa-search xs primary"
    />
    <h4>
      Search
    </h4>
  </div>
  <input
    className="Search an array of classes"
    onChange={[Function]}
  />
</div>
`;
